kie-server-parent/kie-server-tests/kie-server-integ-tests-jbpm/src/test/java/org/kie/server/integrationtests/jbpm/RollingUpdateProcessServiceIntegrationTest.java

```diff
@@ -254,4 +275,178 @@ public void testStartProcessInDifferentDeploymentWithContainerId() throws Except
+                "1.0.1.Final");
+        KieContainerResource containerResource = new KieContainerResource(CONTAINER_ID_101, differentReleaseId101);
+        containerResource.setContainerAlias(CONTAINER_ALIAS);
+        client.createContainer(CONTAINER_ID_101, containerResource);
+
+        Long pid = processClient.startProcess(CONTAINER_ALIAS, PROCESS_ID_EVALUATION_2);
+        assertNotNull(pid);
+        assertTrue(pid > 0);
+
+        processInstance = processClient.getProcessInstance(CONTAINER_ALIAS, pid);
+        assertNotNull(processInstance);
+        assertEquals(CONTAINER_ID_101, processInstance.getContainerId());
+
+    }
+
+    @Test
+    public void testContainerIdAsAlias() throws Exception {
+        Long oldPid = processClient.startProcess(CONTAINER_ALIAS, PROCESS_ID_EVALUATION);
+        assertNotNull(oldPid);
+        assertTrue(oldPid > 0);
+
+        ProcessInstance processInstance = processClient.getProcessInstance(CONTAINER_ALIAS, oldPid);
+        assertNotNull(processInstance);
+        assertEquals(CONTAINER_ID, processInstance.getContainerId());
+
+        // Create the second container with alias equal to the first containerId
+        KieContainerResource containerResource = new KieContainerResource(CONTAINER_ID_101, releaseId101);
+        containerResource.setContainerAlias(CONTAINER_ID);
+        client.createContainer(CONTAINER_ID_101, containerResource);
+
+        assertClientException(
+                () -> processClient.startProcess(CONTAINER_ID, PROCESS_ID_EVALUATION_2), 404, "No process available with given id");
+
+        // Instead the old one should be chosen, since it looks for containerId first
+        oldPid = processClient.startProcess(CONTAINER_ID, PROCESS_ID_EVALUATION);
+        assertNotNull(oldPid);
+        assertTrue(oldPid > 0);
+
+        processInstance = processClient.getProcessInstance(CONTAINER_ALIAS, oldPid);
+        assertNotNull(processInstance);
+        assertEquals(CONTAINER_ID, processInstance.getContainerId());
+
+    }
+
+    @Test
+    public void testSignalNewProcessWithAlias() throws Exception {
+        Map<String, Object> params = new HashMap<>();
+        params.put("nullAccepted", false);
+
+        Long oldProcessInstanceId = processClient.startProcess(CONTAINER_ALIAS, PROCESS_ID_SIGNAL_PROCESS, params);
+        assertNotNull(oldProcessInstanceId);
+        assertTrue(oldProcessInstanceId > 0);
+
+        createExtraContainer();
+
+        Long processInstanceId = processClient.startProcess(CONTAINER_ALIAS, PROCESS_ID_SIGNAL_PROCESS_2, params);
+        assertNotNull(processInstanceId);
+        assertTrue(processInstanceId > 0);
+
+        List<String> availableSignals = processClient.getAvailableSignals(CONTAINER_ALIAS, processInstanceId);
+        assertNotNull(availableSignals);
+        assertEquals(2, availableSignals.size());
+        assertTrue(availableSignals.contains("Signal1"));
```

Comment:
Another assertions where you can benefit from AssertJ - which offers nice assertion methods for collections.